### DISCUSSION TITLE
5.x API Stability?

### DISCUSSION THREAD

**Discussion by aristotaloss** [discord-jda/JDA#2156] [Questions and Help] [answered]
Hi all,

For coursework that I'm creating, I want to get a rough idea of how stable 5.x's API is and how much is planned of this to be final. I don't mean in terms of features to be added, but rather APIs that will be removed or refactoring of existing bits.

This would help me steer clear of using APIs in my course material that are likely to change in the near future, given that 5.x is in alpha. I need the 5.x for the slash commands, as that's Discord's preferred way of handling bot commands moving forwads.

Thanks!

**Comment by MinnDevelopment** [comment, marked as answer]
For the most part, the API is stable. The version is alpha because we are making a few breaking changes to class hierarchy and naming conventions. Some things that are gonna change in the near future:

- [Emoji Names/Usage](https://github.com/DV8FromTheWorld/JDA/pull/2117)
- [Channel Getters](https://github.com/DV8FromTheWorld/JDA/pull/2138)
- [Application Permissions](https://github.com/DV8FromTheWorld/JDA/pull/2113) (forced by discord)

You can see all currently slated breaking changes in the [breaking label](https://github.com/DV8FromTheWorld/JDA/labels/type%3A%20breaking) and all planned or tentative changes in the [project board](https://github.com/DV8FromTheWorld/JDA/projects/5).

Slash commands, specifically, have already been reworked a little bit in JDA 5, and are likely gonna stay stable for a while now. Due to the way the API changes over time, it is always possible that breaking changes have to be made. Discord doesn't make changes to the API with consideration for specific library interfaces, so our hands are forced in some cases, like with [context commands](https://github.com/DV8FromTheWorld/JDA/pull/1769#issuecomment-894770285).
